vocabulary/javascript/vartypes.md

```diff
@@ -0,0 +1,37 @@
+# a list of variable types in js
+
+## general
+
+> variables are containers which can be assigned different values the value (or
+> data) is stored in the computer's memory. variables are an essential aspect of
+> js
+
+## naming variables
+
+- variable names are case sensitive
+- variables cannot be named the same as js keywords such as; if, else or new
+- variable names cannot begin with a number
+
+> it is standard practice to use camelCase when naming variables. camelCase is
+> written without spaces. the first word is lowercase and the first letter of
+> every subsequent word is capitalized. examples: yourName firstParDown
+> nameOfCompany
+
+## variable types
+
+### var
+
+> var is the og keyword for creating a variable. because of its limitations it
+> is not used as often these days but is still common in 'legacy' (older) code
+
+### let
+
+> let was introduced in 2015 with the 6th edition of ESMAScript (ES6 for short).
+> it is a more versatile keyword than var and is used by developers today
+
```

Comment:
Watch out when writing things like this. `var` doesn't have limitations and `let` isn't more versatile. 

In fact, `var` and `let` do exactly the same, but are differently scoped. 
Scoping basically means: "Where are my variables available to me?"
`var` if function scoped and `let` is block scoped.

You can read more about let, const and var here: https://wesbos.com/javascript/01-the-basics/variables-and-statements/#scoping